### PR TITLE
Prepare release script for patches

### PR DESCRIPTION
**Pull request by jtoar**
Partially closes https://github.com/redwoodjs/redwood/issues/2067. This PR prepares the release script for releasing patches.

Also, you can now pass options to answer some of the questions from the CLI and speed things up:
```
yarn release --semver minor --no-prs --checkout
```
But "check" steps will never be skipped.

@thedavidprice, a couple things:

-> I've seen two syntaxes for checking out the release branch:
```shell
# 1
git checkout -b release/patch/v0.28.1 v0.28.0
# 2
git checkout tags/v0.28.0 -b release/patch/v0.28.1
```
I'm going with the first one because that's what's in the git-tagging docs: https://git-scm.com/book/en/v2/Git-Basics-Tagging (at the end—"Checking out Tags").

-> This script now handles updating `next-release-patch` pull requests.

-> This one's probably obvious but we probably have to do a run through to get this 100% right.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "publish:canary": "lerna publish --force-publish --canary --include-merged-tags --preid canary --dist-tag canary --yes --loglevel verbose",
     "release": "node ./tasks/release/cli.mjs",
     "test": "lerna run test --stream -- --colors --maxWorkers=4",
-    "test-release": "NODE_OPTIONS=--experimental-vm-modules ./node_modules/.bin/jest --config ./tasks/release/jest.config.mjs"
+    "test:release": "NODE_OPTIONS=--experimental-vm-modules ./node_modules/.bin/jest --config ./tasks/release/jest.config.mjs"
```
Comment (jtoar) [inline package.json:26]: I'm not sure why I named it `test-release` when the convention for scoping scripts is to use a colon.

Also, I don't have any plans of adding this to CI, but I do run the tests locally while developing.

Comment (thedavidprice) [reply package.json:26]: Maybe this should be `test:release-script` as it's not testing the actual release (or just something else more descriptive).

**Comment by thedavidprice** [comment]
@jtoar I didn't quite follow you here:

> -> I've seen two syntaxes for checking out the release branch:
> 
> # 1
> `git branch release/patch/v0.28.1 v0.28.0`
> # 2
> `git checkout tags/v0.28.0 -b release/patch/v0.28.1`
> 
> I'm going with the first one because that's what's in the git-tagging docs: https://git-scm.com/book/en/v2/Git-Basics-Tagging (at the end—"Checking out Tags").

You are saying we should try `git branch ...`

Just looked through the docs and saw this, which implies `git checkout ...` for me:
Screen Shot 2022-02-07 at 8 05 59 AM

I haven't tried `git branch ...` yet. No preference by me if the results are the same. TBD

**Comment by thedavidprice** [comment]
@jtoar Ah, that makes sense! Thanks.

**Review by Thompson1985 — COMMENTED** [review]
Looks good

**Pull request merged**